### ISSUE TITLE
COE | Accommodate password-protected PDF uploads

### ISSUE THREAD

**Issue by data-doge** [department-of-veterans-affairs/va.gov-team#52022]
## Issue Description

We already do this for NOD, SC, CST, and 526. We don't do it for COE yet.

See https://github.com/department-of-veterans-affairs/vets-website/pull/14610/ and https://github.com/department-of-veterans-affairs/vets-api/blob/7f060a9c999c35fc84669c53ce8458d3fe45a946/app/models/form_attachment.rb#L38 for reference.

---
## Tasks

- [x] Collect password for password-protected PDFs on the FE (using existing tools) so that BE can unlock PDF before storing.
- [ ] We may also have to standardize the way we upload files for the COE form.† 

† There is currently a variation between how we upload docs _during_ the form and _after_ form submission. During the form, we leverage a code path that would automatically unlock PDFs, and after form submission, we leverage a different code path that doesn't automatically unlock PDFs.

## Acceptance Criteria
- [ ] Vet can enter password for password-protected PDFs so that the backend can store the unlocked PDF.

---

**Comment by Midge-dev** [comment]
@tblackwe this appears to have already been implemented to some degree. (I logged into staging w/ 228)
[https://staging.va.gov/housing-assistance/home-loans/request-coe-form-26-1880/upload-supporting-documents](https://staging.va.gov/housing-assistance/home-loans/request-coe-form-26-1880/upload-supporting-documents)
<img src="https://images.zenhubusercontent.com/64679f5ee8120a31820a0a33/fed9958c-f430-45fe-a2ca-0667f4661bda" width="500">

**Comment by Midge-dev** [comment]
Notes:
- At present, FE is capturing PDF passwords on COE
- Payload is being appended on POST w/ that password

Other Discovery:
- Upload URL is `${environment.API_URL}/v0/claim_attachments` and according to Swagger documentation, this doesn't exist anymore. We need to confirm this URL is correct

cc: @tblackwe @saderagsdale @steele-lm

**Comment by Mottie** [comment]
IIRC, `/v0/claim_attachments` was used by 526 to upload files to an S3 bucket. Which the backend would then in turn use to upload the file from the bucket to the app endpoint upon submission. I didn't know that endpoint disappeared; if it did, then it's effecting multiple apps: Burials, pension, COE & 686C-674.

**Comment by Mottie** [comment]
@kylesoskin and I have verified that the endpoint is still working as expected. It's not documented in the swagger docs for some reason 😞

**Comment by steele-lm** [comment]
Sean confirming that FE is working correctly before handing off to backend.

**Comment by micahaspyr** [comment]
Pensions did not fix this, but I have identified the problem.

I was running into user issues as I was getting a "no edipi" error.  The reason was due to a flipper feature being turned off. "profile_user_claims" feature flag must be turned on in the flipper/features page.  Coding looks fairly straightforward.  Looking at specs for test coverage.

It's likely that the fix can be applied to other ndbex forms so I expect it to be future proofed
